src/includes/control.h

```diff
@@ -0,0 +1,107 @@
+#ifndef UI_NODE_CONTROL
+#define UI_NODE_CONTROL 1
+
+#include "ui.h"
+
+#define DEFINE_EVENT(NAME)                                                     \
+  private:                                                                     \
+	nbind::cbFunction *NAME##Callback = NULL;                                  \
+                                                                               \
+  public:                                                                      \
+	void NAME(nbind::cbFunction &cb);
+
+#define IMPLEMENT_EVENT(CLASS, WIDGET, NAME, LIBUI_FUN)                        \
+	static void CLASS##_##NAME(WIDGET *w, void *data) {                        \
+		nbind::cbFunction *cb = (nbind::cbFunction *)data;                     \
+		(*cb)();                                                               \
+	}                                                                          \
+	void CLASS::NAME(nbind::cbFunction &cb) {                                  \
+		NAME##Callback = new nbind::cbFunction(cb);                            \
+		LIBUI_FUN((WIDGET *)getHandle(), CLASS##_##NAME, NAME##Callback);      \
+	}
+
+#define DISPOSE_EVENT(NAME)                                                    \
+	if (NAME##Callback != nullptr) {                                           \
+		delete NAME##Callback;                                                 \
+		NAME##Callback = nullptr;                                              \
+	}
+
+#define DEFINE_CONTROL_METHODS()                                               \
```

Comment:
DEFINE_CONTROL_METHODS, INHERITS_CONTROL_METHODS, DECLARE_CHILD_CONTROL_METHODS, DECLARE_CONTROL_METHODS are not needed anymore, since nbind correctly handles class inheritance by now.